src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -774,28 +774,9 @@ private async Task StartContainerAsync(IExecutionContext executionContext, Conta
                         var node20 = container.TranslateToContainerPath(Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Externals), NodeHandler.Node20_1Folder, "bin", $"node{IOUtil.ExeExtension}"));
 
                         string node20TestCmd = $"bash -c \"{node20} -v\"";
-                        List<string> nodeInfo = await DockerExec(executionContext, container.ContainerId, node20TestCmd, noExceptionOnError: true);
-                        if (nodeInfo.Count > 0)
-                        {
-                            foreach(var nodeInfoLine in nodeInfo)
-                            {
-                                // detect example error from node 20 attempting to run on Ubuntu18:
-                                // /__a/externals/node20/bin/node: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.27' not found (required by /__a/externals/node20/bin/node)
-                                // /__a/externals/node20/bin/node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /__a/externals/node20/bin/node)
-                                // /__a/externals/node20/bin/node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.25' not found (required by /__a/externals/node20/bin/node)
-                                if(nodeInfoLine.Contains("version `GLIBC_2.28' not found")
-                                    || nodeInfoLine.Contains("version `GLIBC_2.25' not found")
-                                    || nodeInfoLine.Contains("version `GLIBC_2.27' not found"))
-                                {
-                                    executionContext.Debug($"GLIBC error found executing node -v; setting NeedsNode16Redirect: {nodeInfoLine}");
-                                    executionContext.Warning($"The container operating system doesn't support Node20. Using Node16 instead. " +
-                                                "Please upgrade the operating system of the container to ensure compatibility with Node20 tasks: " +
-                                                "https://github.com/nodesource/distributions");
-                                                        
-                                    container.NeedsNode16Redirect = true;
-                                }
-                            }
-                        }
+                        List<string> nodeVersionOutput = await DockerExec(executionContext, container.ContainerId, node20TestCmd, noExceptionOnError: true);
```

Comment:
Why do we put `noExceptionOnError: true`, looks that it will throw invalidOperationException in https://github.com/microsoft/azure-pipelines-agent/blob/0780e05b8fde4dd4a6b1f24a09ffce9a948384af/src/Agent.Worker/ContainerOperationProvider.cs#L952

Reply:
nvm, i think i get it.